azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -23,30 +23,6 @@ func resourceArmKubernetesCluster() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
-		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
-			if v, exists := diff.GetOk("network_profile"); exists {
-				rawProfiles := v.([]interface{})
-				if len(rawProfiles) == 0 {
-					return nil
-				}
-
-				// then ensure the conditionally-required fields are set
-				profile := rawProfiles[0].(map[string]interface{})
-				networkPlugin := profile["network_plugin"].(string)
-
-				if networkPlugin == "kubenet" {
```

Comment:
we can also invert this logic to ensure they're not set when the networkPlugin is "azure"

Reply:
Both `azure` and `kubenet` need to be taken care.

---
In reply to: [207462580](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r207462580) [](ancestors = 207462580)